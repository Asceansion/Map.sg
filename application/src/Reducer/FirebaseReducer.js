import { SIGN_OUT_SUCCESS, SIGN_IN_SUCCESS, SIGN_UP_SUCCESS, SIGN_UP_FAILURE, SIGN_IN_FAILURE, SIGN_OUT_FAILURE, CLEAR_ERROR_MESSAGE } from '../Constants/actionTypes';

/**
 * Home reducers to update states belonging to Home view
 * 
 * @Koh Tong Liang
 * @Version 1
 * @Since 31/10/2020
*/

const initialState = {
    loggedIn: false,
    user: null,
    errorMessage: '',
}

function FirebaseReducer (state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case SIGN_IN_SUCCESS:
            return Object.assign({}, state, {
                loggedIn: true,
                user: action.data,
            });
        case SIGN_OUT_SUCCESS:
            return initialState;
        case SIGN_UP_SUCCESS:
            return Object.assign({}, state, {
                loggedIn: true,
                user: action.data,
            });
        case SIGN_IN_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        case SIGN_OUT_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        case SIGN_UP_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        case CLEAR_ERROR_MESSAGE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage
            });
        default:
    } // end of switch case

    return state;
}

export default FirebaseReducer;