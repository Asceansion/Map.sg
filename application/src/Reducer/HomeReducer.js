// import action types
import {
    GET_USER_LOCATION_SUCCEEDED, HISTORY_VIEW, TOGGLE_ROUTE_PLANNER,
    TOGGLE_TRAFFIC_IMAGES_VIEW, TOGGLE_MAP_PICKER, RETURN_MAP_PICKER_RESULT,
    BOOKMARK_VIEW
} from '../Constants/actionTypes';

// Initialise initial values for state
const initialState = {
    routePlannerView: false,
    trafficImagesView: false,
    userLocation: [],
    historyView: false,
    bookmarkView: false,
    mapPickerMode: false,
    mapPickerResult: [],
}

/**
 * Home reducers to update states belonging to Home view
 * @author Koh Tong Liang
 * @version 1
 * @since 31/10/2020
*/
function HomeReducer(state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case 'OVERRIDE_USER_LOCATION':
            return Object.assign({}, state, {
                userLocation: initialState.userLocation.concat({ lng: action.payload.lng, lat: action.payload.lat })
            });
        case GET_USER_LOCATION_SUCCEEDED:
            return Object.assign({}, state, {
                userLocation: initialState.userLocation.concat(action.payload)
            });
        case TOGGLE_ROUTE_PLANNER:
            return Object.assign({}, state, {
                routePlannerView: !state.routePlannerView
            });
        case TOGGLE_MAP_PICKER:
            return Object.assign({}, state, {
                mapPickerMode: !state.mapPickerMode
            });
        case HISTORY_VIEW:
            return Object.assign({}, state, {
                historyView: !state.historyView
            });
        case BOOKMARK_VIEW:
            return Object.assign({}, state, {
                bookmarkView: !state.bookmarkView
            });
        case RETURN_MAP_PICKER_RESULT:
            return Object.assign({}, state, {
                mapPickerResult: initialState.mapPickerResult.concat(action.payload),
            });
        case TOGGLE_TRAFFIC_IMAGES_VIEW:
            break;
        default:
    } // end of switch case

    return state;
}

export default HomeReducer;