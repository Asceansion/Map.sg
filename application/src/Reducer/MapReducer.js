import { GET_TRAFFIC_IMAGES_SUCCEEDED, GET_ERP_DATA_SUCCEEDED, UPDATE_STEPS, UPDATE_CAMERA_MARKERS } from '../Constants/actionTypes';
/**
 * Map Reducer serves to update the state as certain actions are
 * 
 * @Koh Tong Liang
 * @Version 1
 * @Since 30/10/2020
*/

const initialState = {
    cameras: [],
    cameraMarkers: [],
    ERP: [],
    stepNo: [],
    routeInstruction: [],
}

export default function MapReducer (state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case UPDATE_CAMERA_MARKERS:
            return Object.assign({}, state, {
                cameraMarkers: initialState.cameraMarkers.concat(action.payload)
            });
        case GET_TRAFFIC_IMAGES_SUCCEEDED:
            return Object.assign({}, state, {
                cameras: initialState.cameras.concat(action.payload)
            });
        case GET_ERP_DATA_SUCCEEDED:
            return Object.assign({}, state, {
                ERP: initialState.ERP.concat(action.payload)
            });
        case UPDATE_STEPS:
            return Object.assign({}, state, {
                stepNo: state.stepNo.concat(action.payload)
            });
        default:
    }

    return state;
}