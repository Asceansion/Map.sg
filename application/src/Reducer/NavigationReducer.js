import { SEARCH_START_LOCATION_SUCCEEDED, SEARCH_END_LOCATION_SUCCEEDED, PROCESS_START_LOCATION,
    PROCESS_END_LOCATION, PLAN_ROUTE_SUCCEEDED, TRIP_SUMMARY, SAVE_TRIP, MAP_MATCHING_SUCCEEDED, CANCEL_ROUTE, UPDATE_STEPS } from '../Constants/actionTypes';

/**
 * Home reducers to update states belonging to Home view
 * @Koh Tong Liang
 * @Version 1
 * @Since 31/10/2020
*/

const initialState = {
    startLocation: [],
    endLocation: [],
    startLocationSearchResult: [], // for routeplanner form
    endLocationSearchResult: [], // for routeplanner form
    navigationRoute: [], // geometric data of the route user is taking
    mapMatchedRoute: [], // fixed route linestring
    tripSummaryView: false, // toggle view for end of trip summary
    onRoute: false, // show that user is on journey
    stepNo: 0, // indiciate which stage of step by step instruction user is in
    routeInstruction: [], // stores step by step instruction
}

function NavigationReducer (state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case UPDATE_STEPS:
            return Object.assign({}, state, {
                stepNo: initialState.stepNo + action.payload,
            });
        case SEARCH_START_LOCATION_SUCCEEDED:
            return Object.assign({}, state, {
                startLocationSearchResult: state.startLocationSearchResult.concat(action.payload)
            });
        case SEARCH_END_LOCATION_SUCCEEDED:
            return Object.assign({}, state, {
                endLocationSearchResult: state.endLocationSearchResult.concat(action.payload)
            });
        case PROCESS_START_LOCATION:
            return Object.assign({}, state, {
                startLocationSearchResult: initialState.startLocationSearchResult,
                startLocation: state.startLocation.concat({lng: action.payload.geometry.coordinates[0], lat: action.payload.geometry.coordinates[1]}),
            });
        case PROCESS_END_LOCATION:
            return Object.assign({}, state, {
                endLocationSearchResult: initialState.endLocationSearchResult,
                endLocation: state.endLocation.concat({lng: action.payload.geometry.coordinates[0], lat: action.payload.geometry.coordinates[1]}),
            });
        case PLAN_ROUTE_SUCCEEDED:
            return Object.assign({}, state, {
                navigationRoute: state.navigationRoute.concat(action.payload.route),
                onRoute: !state.onRoute,
                routeInstruction: state.routeInstruction.concat(action.payload.routeInstruction),
            });
        case MAP_MATCHING_SUCCEEDED:
            return Object.assign({}, state, {
                mapMatchedRoute: state.mapMatchedRoute.concat(action.payload),
            });
        case TRIP_SUMMARY:
            return Object.assign({}, state, {
                ...initialState,
                tripSummaryView: !state.tripSummaryView
            });
        case CANCEL_ROUTE:
            // return initialState;
            return Object.assign({}, state, {
                ...initialState,
                navigationRoute: initialState.navigationRoute,
            });
        case SAVE_TRIP:
            // store navigation route
            break;
        default:
    } // end of switch case
    return state;
}

export default NavigationReducer;