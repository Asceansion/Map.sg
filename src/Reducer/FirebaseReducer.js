// import action types
import {
    SIGN_OUT_SUCCESS, SIGN_IN_SUCCESS, SIGN_UP_SUCCESS, SIGN_UP_FAILURE, SIGN_IN_FAILURE,
    SIGN_OUT_FAILURE, CLEAR_ERROR_MESSAGE, SAVE_HISTORY_SUCCESS, LOAD_HISTORY_SUCCESS
} from '../Constants/actionTypes';

// Initialise initial values for state
const initialState = {
    loggedIn: false,
    user: null,
    errorMessage: '',
    signInSuccess: false,
    signUpSuccess: false,
    history: [],
}

/**
 * Firebase reducers to update states pertaining to states related to user accounts and data storage
 * @author Koh Tong Liang
 * @version 1
 * @since 31/10/2020
*/
function FirebaseReducer(state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case SAVE_HISTORY_SUCCESS:
            // return Object.assign({}, state, {
            //     history: state.history.concat(action.doc)
            // });
            return;
        case LOAD_HISTORY_SUCCESS:
            return Object.assign({}, state, {
                history: initialState.history.concat(action.payload),
            });
        case SIGN_IN_SUCCESS:
            return Object.assign({}, state, {
                loggedIn: true,
                signInSuccess: !state.signInSuccess,
                user: action.data,
            });
        case SIGN_OUT_SUCCESS:
            return initialState;
        case SIGN_UP_SUCCESS:
            return Object.assign({}, state, {
                loggedIn: true,
                signUpSuccess: !state.signUpSuccess,
                user: action.data,
            });
        case SIGN_IN_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        case SIGN_OUT_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        case SIGN_UP_FAILURE:
            return Object.assign({}, state, {
                errorMessage: initialState.errorMessage.concat(action.payload)
            });
        default:
    } // end of switch case

    return state;
}

export default FirebaseReducer;