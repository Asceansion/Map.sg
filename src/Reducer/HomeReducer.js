import { GET_USER_LOCATION_SUCCEEDED, HISTORY_VIEW, TOGGLE_ROUTE_PLANNER,
    TOGGLE_TRAFFIC_IMAGES_VIEW, TOGGLE_MAP_PICKER, RETURN_MAP_PICKER_RESULT,
    BOOKMARK_VIEW } from '../Constants/actionTypes';
/**
 * Home reducers to update states belonging to Home view
 * 
 * @Koh Tong Liang
 * @Version 1
 * @Since 31/10/2020
*/

const initialState = {
    routePlannerView: false,
    trafficImagesView: false,
    userLocation: [],
    historyView: false,
    bookmarkView: false,
    mapPickerMode: false,
    mapPickerResult: [],
}

function HomeReducer (state = initialState, action) {
    // determine what action to perform and which state to update
    switch (action.type) {
        case 'OVERRIDE_USER_LOCATION':
            return Object.assign({}, state, {
                userLocation: initialState.userLocation.concat({lng: action.payload.lng, lat: action.payload.lat})
            });
        case GET_USER_LOCATION_SUCCEEDED:
            console.log(action.payload)
            return Object.assign({}, state, {
                userLocation: initialState.userLocation.concat(action.payload)
            });
        case TOGGLE_ROUTE_PLANNER:
            return Object.assign({}, state, {
                routePlannerView: !state.routePlannerView
            });
        case TOGGLE_MAP_PICKER:
            return Object.assign({}, state, {
                mapPickerMode: !state.mapPickerMode
            });
        case HISTORY_VIEW:        
            return Object.assign({}, state, {
                historyView: !state.historyView
            });
        case BOOKMARK_VIEW:
            return Object.assign({}, state, {
                bookmarkView: !state.bookmarkView
            });
        case RETURN_MAP_PICKER_RESULT:
            return Object.assign({}, state, {
                mapPickerResult: initialState.mapPickerResult.concat(action.payload),
            });
        case TOGGLE_TRAFFIC_IMAGES_VIEW:
            break;
        default:
    } // end of switch case

    return state;
}

export default HomeReducer;