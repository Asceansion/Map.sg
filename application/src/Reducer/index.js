import { combineReducers } from 'redux';
import HomeReducer from './HomeReducer';
import NavigationReducer from './NavigationReducer';

/* *
 * This is the root reducers, it uses combineReducers to group multiple reducers together
 * and export all of them at 1 go. This serves as entry point to all the other reducers.
 * 
 * @Koh Tong Liang
 * @Version 1.0
 * @Since 30/10/2020
 * */

export default combineReducers({ HomeReducer, NavigationReducer });
