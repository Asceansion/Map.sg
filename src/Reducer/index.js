import { combineReducers } from 'redux';
import HomeReducer from './HomeReducer';
import NavigationReducer from './NavigationReducer';
import MapReducer from './MapReducer';
import FirebaseReducer from './FirebaseReducer';

/* *
 * This is the root reducers, it uses combineReducers to group multiple reducers together
 * and export all of them at 1 go. This serves as entry point to all the other reducers.
 * @author Koh Tong Liang
 * @version 1.0
 * @since 30/10/2020
 * */

export default combineReducers({ 
    HomeReducer,
    NavigationReducer,
    MapReducer,
    FirebaseReducer,
});
