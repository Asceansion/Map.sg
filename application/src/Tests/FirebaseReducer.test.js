
import React from "react";
import { shallow, mount } from "enzyme";
import FirebaseReducer from '../Reducer/FirebaseReducer'
import {FirebaseState} from '../Reducer/FirebaseReducer'

describe('FirebaseReducer ', () => {
    test('should return the initial state', () => {
      expect(FirebaseReducer(undefined, {})).toMatchSnapshot()
    })
  
    test('should handle LOAD_HISTORY_SUCCESS', () => {
      expect(
        FirebaseReducer(FirebaseState,
        {
          type: 'LOAD_HISTORY_SUCCESS'
        })
      ).toMatchSnapshot()
    })
  
    test('should handle SIGN_IN_SUCCESS', () => {
      expect(
        FirebaseReducer(FirebaseState,
        {
          type: 'SIGN_IN_SUCCESS'
        })
      ).toMatchSnapshot()
    })
  
    test('should handle SIGN_OUT_SUCCESS', () => {
      expect(
        FirebaseReducer(FirebaseState,
        {
          type: 'SIGN_OUT_SUCCESS'
        })
      ).toMatchSnapshot()
    })

    test('should handle SIGN_UP_SUCCESS', () => {
        expect(
          FirebaseReducer(FirebaseState,
          {
            type: 'SIGN_UP_SUCCESS'
          })
        ).toMatchSnapshot()
      })

      test('should handle SIGN_IN_FAILURE', () => {
        expect(
          FirebaseReducer(FirebaseState,
          {
            type: 'SIGN_IN_FAILURE'
          })
        ).toMatchSnapshot()
      })

      test('should handle SIGN_OUT_FAILURE', () => {
        expect(
          FirebaseReducer(FirebaseState,
          {
            type: 'SIGN_OUT_FAILURE'
          })
        ).toMatchSnapshot()
      })
      test('should handle SIGN_UP_FAILURE', () => {
        expect(
          FirebaseReducer(FirebaseState,
          {
            type: 'SIGN_UP_FAILURE'
          })
        ).toMatchSnapshot()
      })
      test('should handle CLEAR_ERROR_MESSAGE', () => {
        expect(
          FirebaseReducer(FirebaseState,
          {
            type: 'CLEAR_ERROR_MESSAGE'
          })
        ).toMatchSnapshot()
      })
  })