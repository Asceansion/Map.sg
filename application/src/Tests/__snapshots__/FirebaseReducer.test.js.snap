// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirebaseReducer  should handle CLEAR_ERROR_MESSAGE 1`] = `
Object {
  "errorMessage": "",
  "history": Array [],
  "loggedIn": false,
  "signInSuccess": false,
  "signUpSuccess": false,
  "user": null,
}
`;

exports[`FirebaseReducer  should handle LOAD_HISTORY_SUCCESS 1`] = `
Object {
  "errorMessage": "",
  "history": Array [
    undefined,
  ],
  "loggedIn": false,
  "signInSuccess": false,
  "signUpSuccess": false,
  "user": null,
}
`;

exports[`FirebaseReducer  should handle SIGN_IN_FAILURE 1`] = `
Object {
  "errorMessage": "undefined",
  "history": Array [],
  "loggedIn": false,
  "signInSuccess": false,
  "signUpSuccess": false,
  "user": null,
}
`;

exports[`FirebaseReducer  should handle SIGN_IN_SUCCESS 1`] = `
Object {
  "errorMessage": "",
  "history": Array [],
  "loggedIn": true,
  "signInSuccess": false,
  "signUpSuccess": true,
  "user": undefined,
}
`;

exports[`FirebaseReducer  should handle SIGN_OUT_FAILURE 1`] = `
Object {
  "errorMessage": "undefined",
  "history": Array [],
  "loggedIn": false,
  "signInSuccess": false,
  "signUpSuccess": false,
  "user": null,
}
`;

exports[`FirebaseReducer  should handle SIGN_OUT_SUCCESS 1`] = `
Object {
  "errorMessage": "",
  "history": Array [],
  "loggedIn": false,
  "signInSuccess": false,
  "signUpSuccess": false,
  "user": null,
}
`;

exports[`FirebaseReducer  should handle SIGN_UP_FAILURE 1`] = `
Object {
  "errorMessage": "undefined",
  "history": Array [],
  "loggedIn": false,
  "signInSuccess": false,
  "signUpSuccess": false,
  "user": null,
}
`;

exports[`FirebaseReducer  should handle SIGN_UP_SUCCESS 1`] = `
Object {
  "errorMessage": "",
  "history": Array [],
  "loggedIn": true,
  "signInSuccess": true,
  "signUpSuccess": false,
  "user": undefined,
}
`;

exports[`FirebaseReducer  should return the initial state 1`] = `
Object {
  "errorMessage": "",
  "history": Array [],
  "loggedIn": false,
  "signInSuccess": false,
  "signUpSuccess": false,
  "user": null,
}
`;
